cloud/deploy/bundle.go

```diff
@@ -0,0 +1,242 @@
+package deploy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	astrocore "github.com/astronomer/astro-cli/astro-client-core"
+	astroplatformcore "github.com/astronomer/astro-cli/astro-client-platform-core"
+	"github.com/astronomer/astro-cli/cloud/deployment"
+	"github.com/astronomer/astro-cli/config"
+	"github.com/astronomer/astro-cli/pkg/fileutil"
+	"github.com/astronomer/astro-cli/pkg/git"
+	"github.com/sirupsen/logrus"
+)
+
+type DeployBundleInput struct {
+	BundlePath   string
+	MountPath    string
+	DeploymentID string
+	BundleType   string
+	Description  string
+	Wait         bool
+}
+
+func DeployBundle(input *DeployBundleInput, platformCoreClient astroplatformcore.CoreClient, coreClient astrocore.CoreClient) error {
+	c, err := config.GetCurrentContext()
+	if err != nil {
+		return err
+	}
+
+	// get the current deployment so we can check the deploy is valid
+	currentDeployment, err := deployment.CoreGetDeployment(c.Organization, input.DeploymentID, platformCoreClient)
```

Comment:
Seems like we are doing get deployment twice in this flow...Once in cmd/cloud/dbt.go and again in this function call.